lib/number.js

```diff
@@ -53,12 +58,31 @@ internals.Number = class extends Any {
 
     multiple(base) {
 
-        Hoek.assert(Hoek.isInteger(base), 'multiple must be an integer');
-        Hoek.assert(base > 0, 'multiple must be greater than 0');
+        const isRef = Ref.isRef(base);
+
+        return this._test('multiple', base, (value, state, options) => {
+
+            let divisor;
+            if (isRef) {
+                divisor = base(state.parent, options);
+                if (typeof divisor !== 'number') {
+                    return this.createError('number.ref', { ref: base.key }, state, options);
+                }
+
+                if (divisor === 0) {
+                    return this.createError('number.refZero', { ref: base.key }, state, options);
+                }
+            }
+            else {
+                divisor = base;
+                if ((typeof divisor !== 'number') || (divisor === 0)) {
+                    return this.createError('number.illegalMultiple', { ref: base.key }, state, options);
+                }
+            }
 
-        return this._test('multiple', base, function (value, state, options) {
+            const mult = Math.pow(10, Math.max(internals.precision(value), internals.precision(divisor)));
```

Comment:
@Marsup 
It is to allow number.multiple() to allow decimal numbers. However % operator cannot handle decimal numbers correctly. So the idea is to change the decimal numbers to integers before using % operator.

Example:

- To check if 4.523 is a multiple of 1.3, it checks if 4523 (4.523 * 10^3) is a multiple of 1300 (1.3 * 10^3).
- To check if 4.2 is a multiple of 1.05, it checks if 420 (4.2 * 10^2) is a multiple of 105 (1.05 * 10^2)

Added internals.precision(num) to find the number of decimal places in a any number.